docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -591,63 +594,119 @@ plugin libraries, composed of:
-an attestation header when the `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit in its
+### Summary
+
+API function `oe_get_evidence()` supports the values listed below in its
+`format_id` parameter. The output evidence will be prefixed with an
+`oe_attestation_header` if the `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit in its
 `flags` parameter is set.
+* `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION`
+* `OE_FORMAT_UUID_SGX_ECDSA_P256`
+* `OE_FORMAT_UUID_SGX_EPID_LINKABLE`
+* `OE_FORMAT_UUID_SGX_EPID_UNLINKABLE`
+
+API function `oe_verify_evidence()` supports the values listed below in its
+`format_id` parameter.
+* `NULL`:
+  * The input evidence is generated by `oe_get_evidence()`, with
+  the `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit set in its `flags` parameter.
+* `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION`:
+  * The input evidence is generated by `oe_get_evidence()` for format
+  `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION`, with the
+  `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit cleared in its `flags` parameter.
+* `OE_FORMAT_UUID_SGX_ECDSA_P256`:
+  * The input evidence is generated by `oe_get_evidence()` for format
+  `OE_FORMAT_UUID_SGX_ECDSA_P256`, with the
+  `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit cleared in its `flags` parameter.
+* `OE_FORMAT_UUID_LEGACY_REPORT_REMOTE`:
+  * The input evidence is an OE report generated by the legacy API function
+  `oe_get_report()` with the `OE_REPORT_FLAGS_REMOTE_ATTESTATION` flag.
+* `OE_FORMAT_UUID_SGX_QUOTE_ECDSA_P256`:
+  * The input evidence is an SGX ECDSA quote generated by the
+  Intel SGX SDK DCAP library, or the quote-ex library with algorithm ID
+  `SGX_QL_ALG_ECDSA_P256`.
+
+The table below shows the structure of the evidence data for all the supported
+SGX format IDs, as generated by an attester plugin or verified by a verifier
+plugin.
 
 | Format ID | Evidence structure  |
 | -- | - |
-| `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION` | `oe_report_header \|\| SGX_report(hash) \|\| custom_claims` |
-| `OE_FORMAT_UUID_SGX_ECDSA_P256` | `oe_report_header \|\| SGX_ECDSA_quote(hash) \|\| custom_claims` |
-| `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT` | `oe_report_header \|\| SGX_ECDSA_quote(custom_claims)` |
-| `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE` | `SGX_ECDSA_quote(custom_claims)` |
-| `OE_FORMAT_UUID_SGX_EPID_LINKABLE` | `SGX_EPID_linkable_quote(custom_claims)` |
-| `OE_FORMAT_UUID_SGX_EPID_UNLINKABLE` | `SGX_EPID_unlinkable_quote(custom_claims)` |
+| `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION` | `[ oe_attestation_header ] \|\| SGX_report(hash) \|\| custom_claims` |
+| `OE_FORMAT_UUID_SGX_ECDSA_P256` | `[ oe_attestation_header ] \|\| SGX_ECDSA_p256_quote(hash) \|\| custom_claims` |
+| `OE_FORMAT_UUID_SGX_EPID_LINKABLE` | `[ oe_attestation_header ] \|\| SGX_EPID_linkable_quote(custom_claims)` |
+| `OE_FORMAT_UUID_SGX_EPID_UNLINKABLE` | `[ oe_attestation_header ] \|\| SGX_EPID_unlinkable_quote(custom_claims)` |
+| `OE_FORMAT_UUID_LEGACY_REPORT_REMOTE` | `oe_report_header (for remote attestation) \|\| SGX_ECDSA_p256_quote(custom_claims)` |
```

Comment:
Per SIG attestation meeting, I thought we were also going to define:
* `OE_FORMAT_UUID_LEGACY_REPORT_LOCAL`
* `OE_FORMAT_UUID_SGX_QUOTE_EPID_LINKABLE` (would prefer `OE_FORMAT_UUID_RAW_SGX_EPID_LINKABLE`)
* `OE_FORMAT_UUID_SGX_QUOTE_EPID_UNLINKABLE` (would prefer `OE_FORMAT_UUID_RAW_SGX_EPID_UNLINKABLE`)

Independent of the presence of attester/verifier plugin implementation, it sounded from the meeting like these needed to be pre-defined (effectively UUIDs reserved and published by Intel) and should be documented as such.

Reply:
Sorry I did not get as clear an agreement from the meeting. But I'm fine adding the names to the header file with comments that they are not currently supported and there is no plan to add their support. Along the same line, should `OE_FORMAT_UUID_SGX_ECDSA_P384` be listed too?

@dthaler @yentsanglee @MWShan what's your opinion?

Reply:
Only add UUIDs for which there exists code in OE to consume them.  So until there is code that uses P384, don't add the UUID to the header.

Reply:
> Only add UUIDs for which there exists code in OE to consume them. So until there is code that uses P384, don't add the UUID to the header.

Following the same reasoning, `OE_FORMAT_UUID_LEGACY_REPORT_LOCAL`, `OE_FORMAT_UUID_SGX_QUOTE_EPID_LINKABLE`, and `OE_FORMAT_UUID_SGX_QUOTE_EPID_UNLINKABLE` will not be added since there is no code in OE SDK to consume them.

Reply:
> > Only add UUIDs for which there exists code in OE to consume them. So until there is code that uses P384, don't add the UUID to the header.
> 
> Following the same reasoning, `OE_FORMAT_UUID_LEGACY_REPORT_LOCAL`, `OE_FORMAT_UUID_SGX_QUOTE_EPID_LINKABLE`, and `OE_FORMAT_UUID_SGX_QUOTE_EPID_UNLINKABLE` will not be added since there is no code in OE SDK to consume them.

`OE_FORMAT_UUID_LEGACY_REPORT_LOCAL` (used to be `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION`) is used in the OE SDK code and[ local attestation sample](https://github.com/openenclave/openenclave/blob/master/samples/local_attestation/common/attestation.cpp)

Reply:
The proposed new `OE_FORMAT_UUID_LEGACY_REPORT_LOCAL` is different than the existing `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION`. The former format is generated by `oe_get_report()` for local attestation, and is prefixed with an legacy `oe_report_header_t`. It has no support in `oe_verify_evidence()`. The latter is generated by `oe_get_evidence()` and is consumed by `oe_verify_evidence()`.

The raw EPID quotes generated directly by the quote-ex library (not the EPID evidence generated by `oe_get_evidence()`) are not consumed by `oe_verify_evidence()`.